### PR TITLE
NIFI-12506 /nifi-api/flow/metrics endpoint times out if flow is big (support/1.x)

### PR DESCRIPTION
**Pull request by timeabarna**
# Summary
Related to https://github.com/apache/nifi/pull/8158
[NIFI-12506](https://issues.apache.org/jira/browse/NIFI-12506)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by timeabarna** [comment]
Related to https://github.com/apache/nifi/pull/8158

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java [hunk 1]
```diff
@@ -6117,27 +6123,53 @@ protected Collection<AbstractMetricsRegistry> populateFlowMetrics() {
...
         Set<Connection> connections = controllerFacade.getFlowManager().findAllConnections();
-        for (Connection c : connections) {
-            // If a ResourceNotFoundException is thrown, analytics hasn't been enabled
+        Collection<Map<String, Long>> predictions = Collections.synchronizedList(new ArrayList<>());
+
+        final boolean analyticsEnabled = Boolean.parseBoolean(properties.getProperty(NiFiProperties.ANALYTICS_PREDICTION_ENABLED, Boolean.FALSE.toString()));
+
+        if (analyticsEnabled) {
+            if (parallelProcessingThreadPool == null) {
+                parallelProcessingThreadPool = createParallelProcessingThreadPool();
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java:6132]: See same note on the pull request for the main branch. In this case as well, the thread pool should be created when the ServiceFacade class is initialized.

**Review by exceptionfactory — APPROVED** [review]
Thanks again for working through these changes and implementing an alternative solution for the support branch @timeabarna! +1 merging to the support branch.

**Comment by exceptionfactory** [comment]
Merged in https://github.com/apache/nifi/commit/62ff15a8389f8509370f8c5206c053efebf2eac2

**Pull request closed without merging**